### ISSUE TITLE
Huge List Of Feature Requests, Bugs, Fixes, And Other Improvements

### ISSUE THREAD

**Issue by Elshara** [SocialEngine/phpv4-issues#630]
(Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below.)

What are the steps to reproduce this issue?
-------------------------------------------
1. … Too many to list here.
2. … Not applicable in some cases.
3. … Needs complete code rewrite in others.

What happens?
-------------
… Each issue is a one or two sentence per line problem. Details can be ported in comments.

What were you expecting to happen?
----------------------------------
… Consideration of one or more issues in the list.

Any logs, error output, etc?
----------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.) 

```
Paste Log Here
Create widget for recent notifications.
Create widget for friend requests.
Create widget for group invitations.
Create widget for event invitations.
Create widget for recommendations shared with you such as profiles, content and comments you are tagged in.
Make profile widgets portable.
Make quick add widget for all installed module content types member levels have access to.
Separate joined groups and events from events and groups you have created.
Create option to view pending activities such as friend requests, invitations and responses to messages and other site content.
Create option to manage account settings from other places other than the mini menu.
Create an option to have all profile photos you add to the site, appear as an album called profile photos. All other uploaded photos must specify their own title if creating a new album by default.
Create the option to autoplay music in the profile player widget.
Create option to add albums for videos.
Create files feature and add the ability to organize files by file type.
Create html5 player and photo viewer with inline graphics.
Create map integration for location based modules including tags to tag a location in.
Create the ability to add video responses.
Create the ability to rate on all posts.
Create the ability to view other users posts via widget.
Create the ability to position title, description, tags, custom fields, actions, rate, share, comments, likes and other data items in core module view pages.
Create the ability to independently control the display position of search and filter options.
Create calendar widget for browse module pages.
Create tag cloud for module widget.
Create categories widget for module widget.
Create featured content for all modules as widget.
Create the ability to export and import content types.
Create active module widgets.
Create top module widgets.
Create the ability to have all items a module data set has access to, be viewable on their own pages for sort display.
Create the ability to view module meta data.
Create the ability to reorder the types of search data available to list.
Separate the widgets for members, invited members, unconfirmed, pending, banned, left and rsvp members for groups and events.
Add slider widget.
Add the ability to edit the position of widgets on the notifications requests page.
Add the ability to post relational comments to what ever you are currently interacting with for sub interaction capabilities, such as when viewing or responding to current ratings, playback positions, like or dislike comments or other types of reactions to a post.
Allow the ability for members to create their own categories in addition to tags or hashtags for every post.
Have the option to allow public members to comment, rate, like, share, manage and or contribute anonymously on posts.
Allow members to be able to create or edit posts on the same page using inline pop up forms, such as the ones used for confirmation of friend requests, invitations and error messages.
Allow the ability for members to be able to view folders as well as files for modules containing album contents on browse, manage and view pages.
Allow the ability to view liked content for each module.
Include the ability to include recent forum topics and the ability to view recent posts.
Add the ability to control widget layout for the forums module to manage and view forum discussions.
Add links for statistics module.
Add the ability to let members sign up from the home page widget.
Add the ability for members to be prompted to login or register if they are trying to perform an action while signed out.
Add the ability to view received and sent messages on separate pages.
Add the ability to add custom lists for friends, messages, notifications and other requests of personal content organization.
Add the ability to change member profile type.
Add profile status widget.
Add recent comments widget.
Add recent posts widgets.
Add the ability to view photos in original size or resize them to specific dimensions.
Have the capability to add cover photo.
Create the ability to add RSS feeds.
Create the ability to add custom links.
Add the ability for visitors to be able to view custom sorting options for the display of all content on both widgets and pages.
Create design studio for members and administrators for profiles, posts, comments, pages and site wide features alike respectively.
Add the ability for parts of pages and or separate whole pages to be attached or detached using dependencies for widgets and tab containers.
Add the ability for site to have an API.
Add the ability to sort content by module based on what posts or profiles which you've interacted with by action type.
Add the ability to sell multiple items per post in classifieds section.
Create the ability to have quizzes.
Add the ability to create timeline posts and merdge them into activity feeds for every page.
Add the ability to upload files using different types of uploaders.
Import or export contacts from other social media providers.
Make simpler registration routine.
Ad support for rich text multiformatting nested comments with duplication checks and the ability to move comments from one post, position or module to another.
Add the ability to schedule posts and comments and decide who is able to see the comment based on custom values through user mentions.
Add the ability to let members decide on post creation if comments should be enabled at all or not.
Add the ability for members to tag other members in public and private real time notifications and also as well in an on site shout box.
Add the ability for members to view what other online members are currently viewing.
Add the ability for members to convert posts from one module to another and merdge or split posts or collaborate using different post parameters from one to another.
Add support for the symantic web html5 vector graphics and css 3.
Add ability to embed images, music, videos, files, directories and posts from links and feeds.
Add the ability to directly mention a user, post or group of users and posts, as well as networks.
Add hashtag support.
Add reactions support such as emojis, smileys, stickers, custom embeds or images, videos, bulk actions.
Simplify administration panel into 3 categories, pages, content management and settings.
Create inline template editing for all files and other database values.
Manage existing database fields through the administration panel.
Create package manager for the import and export of a social engine 4 database or file structure.
Create birthday widget.
Create event notification reminders.
When someone tags you in a  post, create the ability to tag them back, like an integrated mutual reaction feature.
Have the ability to report all posts and comments.
Have the ability to autoload new content on browse pages.
Have the ability to view full post inline as well as perform all actions relating to the post from browse pages.
Have the ability to count, record and display who has voted what on polls.
Have the ability to count, record and display who has played a video or song.
Add the ability to record, count and display who has viewed this item.
Add the ability to count, view and display who has shared this item.
Add the ability to separate the modules on profile page by navigation links to their respective modules so profiles can load much easier by association to different sub pages.
Have the ability to generate inbound and outbound native RSS capabilities by core.
Create option to display content based on a specific theme.
Have the ability to upload different file types to a post and have them appear in respective module sections as such under such a post's title album name.
Add chat notification sound when in a chat room as a new message appears.
Add a notification sound if a new notification appears.
Add a way for members to customize the display of their content, profile and settings.
Merdge the profile and settings menus.
Replace flash with html5.
Structure headings based on order of importance.
Redesign admin menu so drag and drop menu pull down options are not a thing anymore.
Create sub tabs for navigation menus.
Allow menu builders to be positioned using layout editor as tab containment fields for easier navigation access.
Allow cursor hover or and on mouse over to display items shown under a menu.
Create the ability to have sponsored content.
Create the ability to view all module contributors.
Create the ability to view contributors of this post.
Create the ability to filter what members have interacted based on what they've done to contribute to the current post if any are found.
Create the ability to assign contributors to a role within a post.
Create the ability to allow the public user group to have some site feature availability.
Create flexible module support allowing people to upload or post anything, with custom field support on the creative part of new posts specifically by members.
Allow members to have the ability to add custom fields to posts.
Give members the ability to attach posts created by them, or with the permission of their speicifc user group or friends, to other posts as post responses rather than comments.
Allow people to create custom actions associated with their profile and posts.
Give people the freedom to to create custom albums of site content they've shared so far.
Add the ability to ban and unban members, by locking their content, profile and ability to view the site.
Merdge the statistics and edit sections of individual profiles listed under manage members in admin panel, with bulk actions support.
Merdge categories, global settings, utilities, custom questions and member level settings for all modules.
Add HTML support for all modules.
Add category setting to control what member levels can post there for all modules.
Add the ability to view, edit and or delete discussions in bulk from admin panel.
Add the ability to remove individual photos from admin panel.
Add the ability to remove individual songs from admin panel.
Add the ability to delete individual posts found in groups in admin panel.
Add the ability to remove individual posts from events in admin panel.
Add the ability to assign members or remove members from groups in admin panel.
Add the ability to assign or remove members from events in admin panel.
Add the ability to assign or remove members from networks in admin panel.
Add the ability to assign or remove members from categories in admin panel.
Add the ability to assign or remove members to and from specific tags mentioning their profiles or posts in admin panel.
Add the ability to view all comments based on module in admin panel.
Add the ability to delete and view activity feed posts in admin panel.
Add the ability to monitor notifications in admin panel.
Add the ability to monitor and delete messages in admin panel.
Add the ability to completely edit member settings in admin panel.
Add the ability to merdge private and public chat.
Add the ability to view recent comments for each module.
Add the ability to include status feeds as widget.
Add the ability to include profile updates as widget.
Add the ability to comment on anything using comments module by post url.
Add the ability to comment or like announcements.
Add the ability to interact with ad campaigns.
Add the ability to reorganize module layout and menu orders in the admin menu.
Add the ability to change short cuts in admin panel.
Add the ability to add all admin pages to layout editor and have it be a dedicated administrator feature separate from the rest of the site pages.
Add the ability to control the layout within groups and events.
Add the ability to mirror posts made in groups and events, to other parts of the site using relative modules and vice versa.
Add the ability to control the amount of data sorting by table names in header fields.
Make timeline widget possible specific to global and individual modules.
Make individual module and post statistics widgets.
Allow posts and comments to contain a schedule.
Create mentions for posts and profiles.
Replace ID system with current page URL.
Allow core pages to be edited inline.
Allow modules pages to contain their own custom URL and name.
Allow members to send customizable reactions to one another such as greetings, pokes, gifts or other such things.
Allow members to have a bulk content manager for all their posts and associated data streams.
Include redirection through URL encoding for sign up, sign in and sign out.
Choose where to redirect visitors to upon completion of sign up, sign in and sign out.
Prioritize performance of sign up, sign in and sign out.
Make sign up, sign in and sign out use ajax.
Have Ajax be used when writing a new post.
Create ajax integration for site wide use.
Add sub tab capabilities.
Add redirection widget for all modules.
Add option to automatically check to see if content exists by mentions and links to it using urls. If page was updated, search for current url. If page does not exist, delete broken references automatically.
Add option to download embedded contents to server and use copy of such content to be always available, even if the source of the upload goes offline.
Specify the ability to control how data is sent and received.
Add the ability for members to be able to import and export their data.
Add the ability to connect to any online web API for integration access.
Make follows and friendships mutually distinguished.
Transform networks into a member customizable feature where they can choose specific lists of members they wish to be associated with.
Allow members to see statistics related to their activity on the site not viewable by anyone else except for configured people or user levels.
Allow user levels to have moderator, administrator and statistic roles related to site profiles and contents.
Allow profile questions to be automatically answered using existing data stored within a browser session. Such as the IP of the members current location. The time zone and language the member is using.
Allow back end system to receive third party module packages.
Allow different types of database types to be received upon installation of social engine for dadatabase compatibility.
Give package manager specific permission levels.
Allow all content and profiles to have their own sub domain.
Allow packages to be enabled or disabled from within control panel.
Allow new themes to be created from control panel.
Allow new languages to be created from control panel.
Assign bootstrap to all social engine design.
Allow members to receive notifications at a certain time.
Give better file structure to core and modules.
Assign one file per function for better stability.
Allow members to be promoted to different profile types or member levels depending on site activity.
Fix the friendship statistics area.
Allow members to update the state of blog posts be they published or drafts.
Create email digests highlights for notifications.
Create leaderboards system.
Allow relative actions to be detected in job queue.
 Creating dependencies on database updates as triggers for jobs to be run on request.
Request data from database only if file storage does not have access to it.
Cache all pages respectively unless individual updates are associated with them and or the database itself.
Set custom time out for members log in sessions.
Allow the ability for file storage to contain module settings.
Create spam detection system.
Improve sitemap to create urls for all update pages. As well as member profiles, content and modules.
Generate RSS feeds for inbound and outbound content. Allow members to create their own.
Allow site images, audio, videos and other files to contain an optional water mark.
Create badges system.
Allow an admin password for administration panel.
Allow members to perform actions for set periods of time in member levels.
Allow administrators to specify new privacy groups for member level settings based on member interaction.
Allow body injection for general settings area.
Allow custom splash page integration.
Allow resizing of photos. As well as being able to choose custom widths and heights.
Allow a member to use site anonymously.
Allow a member to have multiple profiles and profile types.
Allow option to display profile photo, profile ID or address, site name for any time a profile is mentioned or a piece of content a member posts.
Customize the number of items profile widgets can display at one time.
Create image GD library integration.
Create CSRF token integration.
Create safe iframe uri integration.
Create html purifyer integration.
Create ignored words for tags when searching.
Create minimum number of tags setting to be included in search results.
Create permalinks integration.
Notify members if any change to their personal level are updated.
Create integration with spam databases.
Allow the ability to edit and create custom cookies.
Move all display settings for core and all modules to their respective widget components in layout editor.
Replace content widget with movable widgets comprised of such material.
Allow specific widgets to be positioned on any page.
Add ability for administrators to write notes in admin panel. And for users to do the same respectively.
Add the ability to manage file attachments.
Add the ability to add categories to contact form.
Add the ability to have custom egifts with categories.
Add the ability to manage individual features and paid access.
Have the ability to promote sponsored content and manage how people are able to mark a sponsorship.
Add custom input fields for all modules. Manage existing fields such as title, description, tags and how they work, what they are related to.
Add archive management support for mass mailing.
Create content transfership system.
Add categories for report system.
Support report functionality on all posts, comments and browse or manage pages.

Support adding custom share sites.
Support core edits to existing structure via third party modules.
Support antispam quiz.
Support relative system page navigation.
Add custom page access rules.
Provide statistics per module about how people have interacted with it.
Provide ability to check unknown files in installation.
Provide ability to check for duplicates.
Provide ability to check modified files of installation.
Provide ability to refresh counters.
Add reparser support.
Add ability to view private messages.
Add the ability to manage and add custom field types. Such as countries, cities, states, postal codes, phone numbers, verified email address domains, websites and other types of third party integration.
Add the ability to view online members as well as guests and bots.
Add the ability to manage the database from the administration panel.
Support custom currencies.
Support rewriting of URLs.
Support custom seo elements.
Add the ability to follow, unfollow, index, not index and perform other SEO htaccess edits inline from administration panel.
Add the ability to add custom settings to existing modules.
Add the ability to find missing settings.
Add emoticon and emoji support.
Add apps support.
Add the ability to extend existing functionality of core and module settings.
Add the ability to have custom user group settings.
Add ability for members to choose a reason why they wish to remove their account.
Add ability to prune old data and inactive members.
Add archive handler support.
Add text prior to image loading support for all modules.
Cache user data, profiles and posts as well as plugin files and settings.
Add live shout box.
Add birthdays support.
Add recently viewed support.
Add profile tracking and post tracking support.
Cache database queries.
Cache all available filter options.
Run cache as a part of module maintenance in task scheduler or cron and or their equivalent.
Keep images resized by website.
Add individual ajax support.
Add custom display of date and time.
Add force 404 check.
Support user IP substring in ID hash.
Add custom wyciwyg, html, bbcode, mark down and other editor integrations.
Replace URL strings with anchor links.
Add the ability to shorten parsed URL links to a custom value.
Add ability to resize embedded and uploaded videos.
Support html and line breaks in activity feed.
Have support for exchange rate API integration.
Add debug support.
Add custom copyright information in general settings of admin panel.
Add custom page title delimiter support.
Support the ability to change the order of page title, description, tags and other custom meta data and make them unique for every page.
Add support for browser compatibility check.
Add launch date support.
Add IP InfoDB support.
Add language flag support if other packages other than English or the current translation are available.
Autodetect browser time zone and language.
Add no string restriction UTF8 and other custom font support such as unicode for language data.
Add default site email signature support.
Add checkdns integration support.
Support the ability to customize multi step registration form.
Add custom minimum, maximum and limitation support for input field data.
Suggest custom names if profile address isn't available.
Choose friendship on user registration.
Choose default privacy setting for users.
Choose default notification settings for users.
Redirection after sign up, sign in and sign out.
Add privacy policy to terms of service confirmation.
Add option to have users retype their email instead of requiring to verify it.
Set custom meta description, tag and title limits.
Add list of ignored keywords.
Include custom page titles for custom field data.
Support url cropping.
Add external link detection.
Add no follow no index options for external links.
Add Warn when visiting an external link support.
Add external or internal link white list and black list support.
Support the ability to Customize the amount of posts a user has to make prior to being able to link something as a trusted member.
Create duplicate content management system.
Provide the ability to force https support.
Add custom IP support to admin control panel.
Add support to authenticate users via a database session log.
Add unique salt for ID hash support.
Add HTML body check for specific text support.
Create active session for logged in users limitation.
Add GZIP compression level support.
Add CSRF protection  support.
Add session IP octet length check support.
Add build directories support for public items organization.
Add log all site activity support.
Cache all java, php, html, css, ajax, database queries and template files.
Add getID3 file support.
Add hash bang disable URL support if ajax browsing is enabled.
Add custom template support when site is in maintenance mode.
Add Akismet  integration support.
Support links in module titles.
Support PM hash check.
Support spam check.
Support spam count protection.
Support removal and deletion of spam content if person considered as spammer meets spam requirements.
Support removal of all external URLs.
Support removal of all external emails.
Add custom time stamp support.
Enable built in captcha support.
Support nested comments.
Support video comments.
Allow multiple people to be notified if someone sends a contact message of the admin level.
Allow multiple to be notified if someone sends an abuse report for the administrators.
Support autoresponses for contact and abuse report messages.
Add option whether to show or hide custom fields.
Support display of custom comment cut off on feeds.
support display of custom likes cut off on feeds.
Add group duplicate feed or relative feed per module support per user.
Support automatic refresh and loading of new and scrolled content in feeds.
Add custom Limits on how far a feed can scroll.
Share updates from feed to facebook and twitter as well as other social profiles via janrain.
Add feeds in the past option to timeline.
Support timeline as well as feed.
Allow a member to choose for timeline and feed.
Add feed comments and likes support.
Support check in by location integration.
Support loading of ajax user content after site core loads.
Support to cache existing user content.
Support active posts in forum for custom time limit.
Support display of custom friend request limit.
Support birthday notices.
Support display of friend suggestions.
Support custom ways to have friend suggestions appear.
Support removing user if denied request from friends pending request list exists.
Support option to allow or deny blocked users to send friend requests or contact you in any other way.
Support mutual friends browsing.
Add support to see if any friends you have are online.
Add custom message support if a user tries to add another user as a friend.
Support inviting of contacts from social media and posting messages, requests and share links directly to them.
Support custom invite time out.
Support the list of pending invitations.
Support duplicate invitation check.
Support member already registered using email address specified in invitation check.
Lift hard 10 user per session invitation limit. Allow it to  be flexible.
Support language development brackets if state of translation is incomplete.
Support the ability to show member photos when they like or comment on an item.
Support dislike.
Support log of user being online for specified custom time limit as active session.
Support custom folders and their count of messages contained within for private mail.
Support total mail count.
Support custom limit of messages. Show warning if close to exceeding such.
Support the inability to a select a member to mail if they have blocked you.
Support update message read on notification preview.
Support option of site wide notification update.
Support custom notification title and description length in preview.
Support the ability to crop photos.
Add theater mode for photos and all other modules.
Add photo battle for photos and all other modules.
Add support to show or hide profile photos in gallery.
Add support to show certain custom number of categories.
Create pokes system.
Add support of pokes to activity feed.
Add complementary image to polls and other modules as a featured design.
Support recent takers for polls.
Add custom profile title.
Create quizzes system.
Add support to show percentage or correct and not correct answers as well as results in percentages.
Support recent takers for quizzes.
Add support for featured picture of quiz.
Create live shout box system.
Support custom display of messages for shout box.
Support word wrap for shout box.
Support archived messages for shout box.
Support tag system and popular search terms.
Support custom display of tags.
Support custom display of buttons on wyciwyg editor.
Support pop up of message informing a user has to be signed in to be able to perform a certain action if signed out or not registered.
Support requirement for privacy check option if a user is at a certain age or not.
Support verification of email time out.
Support logout immediately after changing email or password.
Display if a member is online if you are viewing somewhere their profile is mentioned or if you are on their profile itself.
Add custom date of birth beginning and end values.
Support privacy to hide users if they are at a certain age or not from browse members area.
Add tool tip for statistics about a particular item mentioned, comment, users profile or post.
Create brute force prevention system.
Support attempts count and time out for brute force prevention.
Create activity points system.
Support the ability to allow members flexibility to pay for sponsored products using activity points tied in with promotion system of member levels.
Add support for activity points to be converted to available currency types and vice versa.
Support the ability to ahve members purchase activity points.
Support alternative password reset routine.
Support joining or splitting first and last names.
Support alternative registration step system.
Support option to view the last time a user was on the site.
Support custom embed options for videos.
Support ability to autoplay media such as photo slide shows, music playlists and videos.
Support alternative to ffmpeg.
Support posts and comments to have thumbnails associated with them.
Support ads to be hidden from certain member levels or profile types.
Add administrator and moderator based customization to member level settings.
Support the ability for members to close announcements once read.
Support the ability for members to browse announcements.
Add support for apps to be added by members.
Add file size limit for attachments.
Add ability for members to control who can add and manage attachments.
Add the ability to allow category limits to be imposed on certain member levels for all modules.
Add the ability to allow member level permissions to let users add their own category to any module.
Add the ability for member levels to be able to control specific comments on an item added manually for management assignment to specific users.
Add the ability for members to password protect content.
Add ability for members to add a time limit prior to being able to post specific module type again.
Add ability for member levels to control whether or not content can be searched.
Add ability for member levels to control whether or not content can be liked.
Add ability for member levels to control whether or not content can be rated.
Add ability for member levels to control whether or not content can be disliked
Add ability for member levels to control whether or not content can be reported
Add ability for member levels to control whether or not content can be shared while viewing post.
Add ability for member levels to control whether or not content can use a wyciwyg editor.
Add ability for member levels to control whether or not content can have attachments posted to it.
Add ability for member levels to control whether or not content can be featured.
Add ability for member levels to control whether or not content can be sponsored.
Add ability for member levels to control whether or not content can count towards activity points.
Add ability for member levels to control whether or not content can be edited by you, a member of the same member level, or a member of another member level.
Add ability for member levels to control whether or not content can be deleted by you, by another member in the same level, or by another member in a different level.
Add ability for member levels to control whether or not content can be viewed by you, a member of the same member level, or by a member of another member level.
Add ability for member levels to control whether or not content can be visible at a certain time, or can be commented or responded to at a certain time.
Add ability for member levels to control whether or not content can be copied or duplicated.
Add ability for member levels to control whether or not content can be transferred to someone other than its creator, or somewhere other than its current module or post sequence.
Add ability for member levels to control whether or not content fields can be updated.
Add ability for member levels to control whether or not content can be restored by someone who has left, been banned from, or who has been deleted from the site.
Add ability for member levels to control whether or not content can record the count of interactions and interaction types associated with it, and whether or not such a count can be reset.
Add ability for member levels to control whether or not content has to be approved for participation of interaction by a confirmation system.
Add ability for member levels to control whether or not content can have custom fields associated with each new post shared to its corresponding module.
Add ability for member levels to control whether or not content can be discussed or mentioned by certain people.
Add ability for member levels to control whether or not content can have notifications associated with it for every interaction type allowed to add to it.
Add ability for member levels to control whether or not content can have custom privacy options.
Add ability for member levels to control whether or not content can be dependent on recurring dates or specific times.
Add ability for member levels to control whether or not content requires a verified profile to post to its corresponding module or not.
Add ability for member levels to control whether or not content is suitable for a target audience.
Add ability for member levels to control whether or not content is allowed to remain if it contains verified taboo subject matter or if it should be placed in a queue for further review or possible deletion as further action specifies.
Add ability for member levels to control whether or not content can have automatic actions associated with it set up by the member who owns it.
Support custom captcha integration for all module types.
Support the ability to hide or show update information on items.
Support the ability for specific users to bypass the site maintenance page.
Support member levels who are exempt from spam check.
Support the ability for activity points to be gifted by specific member levels.
Support the ability for custom fields to be associated with a particular member level.
Add ability for member levels to control whether or not members can add service integration to content.
Add ability for member levels to control whether or not members can hide content from other people.
Add ability for member levels to control whether or not members can update the display order of site filters with or without approval.
Add ability for member levels to control whether or not members can change settings with or without approval.
Add ability for member levels to control whether or not categories are restricted to certain people, or change how their functionality behaves.
Add ability for member levels to control whether or not members can mark specific content using custom interactions.
Add ability for member levels to control whether or not members can merdge posts.
Add ability for member levels to control whether or not members can check for duplicate posts.
Add ability for member levels to control whether or not members can use and archive one way or multiple way points of communication with verification.
Support if member levels can add friends.
Support member levels being able to add custom folders with optional limit.
Support member levels being able to remove friends.
Support member levels being able to block others.
Support member levels being able to unblock others.
Support member levels being able to use chat.
Support member levels being able to private message.
Support member levels being able to greet people.
Support member levels being able to play media.
Support member levels being able to vote on content.
Support member levels being able to participate in group or event activities.
Support member levels being able to favorite or unfavorite, like or dislike an item.
Support member levels being able to use the contact form.
Support member levels being able to report abuse.
Support member levels being able to receive notifications.
Support member levels being able to invite people.
Support member levels being able to control who can contact them and who they can contact by private message.
Support member levels being able to control how many people they can contact at once.
Support member levels being able to send themselves private messages.
Support member levels being able to view counts and statistics.
Add the ability to support expired or on hold content.
Support the ability to allow member levels to manage content posted by others in settings such as albums, groups and events.
Support member levels being able to manage mass emails and also manage subscriptions to them in bulk.
Support member levels being able to submit a transfer ownership request for a piece of content or profile.
Support member levels being able to collaborate as one or multiple content owners on a single post.
Support member levels being able to download content.
Support optional limit of how many tags a post or profile can have.
Support the ability to view existing poll results prior to voting.
Support the ability for a member to vote only once, or change their vote.
Support members being able to optionally limit how many answers a poll has.
Support members being able to view who voted what.
Support members being able to vote on their own polls.
Support members being able to edit posts after they are published.
Support members being able to highlight their own posts.
Support the ability for members to set a custom allow list or block list when posting content.
Support the ability to custom design profiles and other content.
Support the ability to add profile cover photos as well as cover photos for other content.
Support album content for blogs, forum topics, polls, classifieds and videos.
Support user created albums based on content people already have posted.
Support members being able to create their own layout theme or template base.
Create quotes system.
Create news system.
Create articles system.
Create wiki system.
Create petition system.
Create pages system.
Create ban system.
Create avatars system with integration with gravatar.
Create feedback system.
Create files system.
Create link directory system.
Create page access control system.
Create site customizer and design studio system.
Create store system.
Create sites system.
Create world map system.
Create spy system.
Create zipcode search system.
Create Google search system.
Create white board write board system.
Create timeline outline system.
Improve chat.
Create poke system.
Create badges system.
Create rank system.
Create reward and or award system.
Create tracking system.
Use alternative media player. Flowplayer. VideoJS. JPlayer. JWPlayer. Kaltura. Etc.
Create URI DNS blocklist system for spam protection. Use integrations such as. Akismet. Stop forum spam.
Create Zend replacement such as Volt or Twig.
Create MYSQL database replacement or compatible system.
Reduce HTTP network round trips.
```

Any other comments?
-------------------
… Some issues may be duplicates of ones already reported or posted by others. Others may be requests of things to come, or bugs currently present. The data is up to date and current however, so no previous versions which bugs have been fixed already, are of concern.

What versions of software are you using?
----------------------------------------
**Operating System:** … Windows 10 personal. Linux web server.

**SocialEngine PHP Version:** … 4.8.13

**Comment by smazurov** [comment]
This request is impossible to process in any meaningful way, so I will close it.
